draft-ietf-quic-multipath.md

```diff
@@ -375,6 +375,10 @@ that address. After receiving packets from the
 client on a new path, if the server decides to use the new path,
 the server MUST perform path validation ({{Section 8.2 of QUIC-TRANSPORT}})
 unless it has previously validated that address.
+An endpoint MAY deny the establishment of a new path initiated by its
+peer by not replying with a PATH_RESPONSE frame but sending a
+PATH_ABANDON frame on another path for the associated Path ID used
+for the PATH_CHALLENGE instead.
```

Comment:
Do we actually need to change this to a MUST: "If an endpoints decides to deny the establishment of a path, it MUST send an PATH_ABANDON frame..."?

Reply:
I tend to support use MUST here, as we have insisted that PATH_ABANDON frame is a necessary explicit signal for path closure. If endpoint doesn't send PATH_ABANDON frame here, the peer may not know what happened with the path, and also not sure whether the path id resource should be consumed or not.

Reply:
I would not use normative language. In fact, we have  potential race conditions: arrival of a new connection ID, availability of other resource, etc. If the endpoint ignores the path challenge, the challenge will be repeated a bit later, and the race conditions will be resolved. I think it is generally better to ignore the path challenge, and let the peer abandon the path when it abandons its own challenge -- unless there is a strong reason, such as "please don't use that interface."

Reply:
Is that better?
```suggestion
If an endpoint wants to deny the establishment of a new path initiated by its
peer by not replying with a PATH_RESPONSE frame, this is considered a path
closure and requires sending a
PATH_ABANDON frame on another path for the associated Path ID used
for the PATH_CHALLENGE as specified in section {{path-close}}.
```

Reply:
@huitema what's the problem of sending an path_abandon immediately rather then waiting for the peer to give up?

Reply:
Race conditions. There are cases when the endpoint will not be ready to accept the PATH CHALLENGE, but will be ready a moment later, when a new PATH CHALLENGE is repeated on the same path.

In normal behavior, the endpoint just ignores the path challenge. The peer may try to repeat it, or may eventually abandon the path, triggering the reciprocal abandon by the challenged endpoint.

Reply:
But that's fine because the texts say "If an endpoint wants to deny the establishment". In you described case it does not want to deny it but it might delay it a bit. So ignoring the path challenges is always okay because there is also nothing you can do to enforce any other behavior. However, if the endpoint already notes that it doesn't want to have the connection then it should perform a path close.

Reply:
Maybe, but I am afraid that this is a bit too subtle, and that implementers will just go for "queue an Abandon" instead of doing nothing. Maybe give a little warning, such as, cannot agree because of missing resource.

Reply:
```suggestion
A server can also decide to not perform path validation if it wants to deny the new path,
or not perform it immediately if, e.g., no connection ID is yet available. 
If an endpoint decides to deny the establishment of a new path,
this should be considered as a path closure and therefore requires sending a
PATH_ABANDON frame on another path for the associated Path ID used
by the client as specified in section {{path-close}}.
```